File: FootScout-Vue.WebAPI/Entities/FavoriteClubAdvertisement.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    public class FavoriteClubAdvertisement
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int ClubAdvertisementId { get; set; }

        [ForeignKey("ClubAdvertisementId")]
        public virtual ClubAdvertisement ClubAdvertisement { get; set; }

        [Required]
        public string UserId { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; }
    }
}

File: FootScout-Vue.WebClient/src/models/interfaces/FavoriteClubAdvertisement.ts
import type { UserDTO } from '../dtos/UserDTO';
import type { ClubAdvertisement } from './ClubAdvertisement';

export interface FavoriteClubAdvertisement {
    id: number;
    clubAdvertisementId: number;
    clubAdvertisement: ClubAdvertisement;
    userId: string;
    user: UserDTO;
}